### PR TITLE
Add new package: marv

### PR DESCRIPTION
**Pull request by rjoonas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## marv ([<kbd>unpkg</kbd>](https://unpkg.com/browse/marv@latest/))
was missing the following properties:
1. migrate
1. scan
1. drop
1. parseDirectives
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 938032216626449deeadeb875216f6f7873899f8
</p>

File: types/marv/api/callback.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/// <reference types="node" />
+
```
Comment (peterblazejewicz) [inline types/marv/api/callback.d.ts:2]: should be ok to use this only in the 'index.d.ts'
```suggestion
```

File: types/marv/api/callback.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import { PathLike } from 'fs';
+import { Driver, EmptyObject, ErrorOnlyCallback, ParsedMigration, ScanOptions } from '../index';
```
Comment (peterblazejewicz) [inline types/marv/api/callback.d.ts:4]: make 'index.d.ts' contain only types (not symbols like const/class), and use:
```ts
import type {...} ...
```
in your submoudles (api/callback, api/promise). The `import type ` syntax is now supported by default on DT repo (TS 3.9 is the lowest version in support version window)

File: types/marv/marv-tests.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import marvDefault, { Driver } from 'marv';
```
Comment (peterblazejewicz) [inline types/marv/marv-tests.ts:1]: IMO it's not a default, it's module re-exports. (that's trivia, but could mislead readers).
it should be possible to get those I think:
```ts
migrate: [Function: migrate],
scan: [Function: scanDirectory],
drop: [Function: drop],
parseDirectives: [Function: parseDirectives]
```

File: types/marv/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// Type definitions for marv 5.0
+// Project: https://github.com/guidesmiths/marv
+// Definitions by: Joonas Rouhiainen <https://github.com/rjoonas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="./api/callback.d.ts" />
+
+import * as callbackApi from './api/callback';
```
[the remark this answers is not in the document]
Comment (peterblazejewicz) [reply types/marv/index.d.ts:8]: add a node TS import directive here to be shared in DT module

File: types/marv/marv-tests.ts (hunk 3 above)
Comment (peterblazejewicz) [reply types/marv/marv-tests.ts:1]: yeah, see dt-bot comment on missing symbols from 'index' scan

**Comment by rjoonas** [comment]
Thank you for the fast review @peterblazejewicz! ⚡ I pushed a fix commit that should address the issues you found. Pretty big mistake on my part there, typing a CJS module as having a default export. Good catch. 👍

File: types/marv/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+// Type definitions for marv 5.0
+// Project: https://github.com/guidesmiths/marv
+// Definitions by: Joonas Rouhiainen <https://github.com/rjoonas>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+/// <reference path="./api/callback.d.ts" />
```
Comment (peterblazejewicz) [inline types/marv/index.d.ts:7]:
```suggestion
```

it's already imported via module import

Comment (peterblazejewicz) [inline types/marv/index.d.ts:7]:
```suggestion
```

it's already imported via module import

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@rjoonas thanks!

**Comment by rjoonas** [comment]
Ready to merge

**Pull request merged**